src/Security/RolesDownloadStrategy.php

```diff
@@ -35,16 +36,34 @@ class RolesDownloadStrategy implements DownloadStrategyInterface
     protected $security;
 
     /**
-     * @var TranslatorInterface
+     * @var LegacyTranslatorInterface|TranslatorInterface
      */
     protected $translator;
 
     /**
      * @param AuthorizationCheckerInterface $security
      * @param string[]                      $roles
      */
-    public function __construct(TranslatorInterface $translator, $security, array $roles = [])
+    public function __construct(object $translator, $security, array $roles = [])
     {
+        if (!$translator instanceof TranslatorInterface) {
+            @trigger_error(sprintf(
+                'Argument 1 passed to "%s()" MUST be an instance of "%s", "%s" given.',
+                __METHOD__,
+                TranslatorInterface::class,
+                \get_class($translator)
```

Comment:
BTW, this deprecation should be placed after the type check performed in line 58, in order to avoid to trigger the error before the exception.
```suggestion
                'Passing other type than "%s" as argument 1 to "%s()" is deprecated since sonata-project/media-bundle 3.x'
                .' and will throw a "\TypeError" error in 4.0.',
                __METHOD__
```

Reply:
I merge two if in one if-else.

Reply:
It's fine, but please update the deprecation in order to be consistent with [the way we usually express these messages](https://github.com/sonata-project/SonataAdminBundle/search?q=%22passing+other+type%22).

Reply:
Done